content/tutorials/reactor-firehose-kinesis.textile

```diff
@@ -0,0 +1,604 @@
+```
+
+In the code you wrote above, you are referencing your Ably API key as an environment variable, @ABLY_API_KEY@. In order for this to work, you need to set this environment variable. Execute the following AWS CLI command to achieve this, replacing @<your api key>@ with your Ably API key:
+
+```[sh]
+aws lambda update-function-configuration --function-name ProcessKinesisRecords --environment Variables={ABLY_API_KEY=<your api key>}
+```
+
+Finally, you need to tell your lambda when to execute. In this case, you want it to run when items appear in your data stream, so you need to configure an event source mapping between your lambda and your stream. Run the following command, replacing the argument to @--event-source@ with the ARN of your Kinesis data stream that you created in Step 1.
+
+```[sh]
+aws lambda create-event-source-mapping --function-name ProcessKinesisRecords --event-source arn:aws:kinesis:<region>:<id>:stream/sensor-data-stream --batch-size 5 --starting-position LATEST
+```
+
+With your lambda function ready to calculate the average temperature and humidity readings from your sensors and publish those results to an Ably channel, you now need to subscribe to that channel so that you can display them.
+
+h2. Step 8: Display the sensor data averages
+
+In the final step you will subscribe to the @processed-sensor-data@ channel and display the messages that your lambda publishes to that channel.
+
+First, you need to authenticate the client with Ably. You will use "token authentication":/core-features/authentication#token-authentication to achieve this. 
+
+In @server.js@, add a new @/auth@ route (just below the home route @/@) that requests a short-lived token from Ably and returns it in the response:
+
+```[javascript]
+app.get("/auth", (request, response) => {
+  ably.auth.requestToken((err, tokenDetails) => {
+    if(err) {
+      console.error(err)
+      response.sendStatus(500)
+      return
+    }
+    response.json(tokenDetails)
+  })
+})
+```
+
+Then, in @client.js@, add the following code under the @generateSensorData()@ function. This code instantiates the Ably client and subscribes to the @processed-sensor-data@ channel:
+
+```[javascript]
+function generateSensorData() {...}
+
+const ably = new Ably.Realtime({ authUrl: "/auth" })
+const channel = ably.channels.get("processed-sensor-data")
+channel.subscribe((msg) => {
+  console.log(msg.data)
+  const data = JSON.parse(msg.data)
+  document.getElementById(
+    "count"
+  ).innerHTML = `<strong>Number of readings: </strong>${data.num_readings}`
+  document.getElementById(
+    "averages"
+  ).innerHTML = `<strong>Average temperature (°C):</strong> ${data.avg_temperature.toFixed(
+    2
+  )}<br/><strong>Average humidity (%):</strong> ${data.avg_humidity.toFixed(
+    1
+  )}<br/>`
+})
+```
+
+h2. Step 9: Try it out
+
+You are now ready to test this!
+
+# Run your server by executing @node server.js@ in the root of your project directory.
```

Comment:
Wasn't sure if these were meant to be numbered steps. In the Next steps section they are numbered.

Reply:
@tbedford Am I missing something here? Do I have to number these manually (as I did in Next steps)?